Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/Model/Building/SqlStorageTypeInformationProvider.cs

```diff
@@ -226,6 +229,42 @@ public IStorageTypeInformation GetStorageType (object? value)
       return null;
     }
 
+    private StorageTypeInformation GetDateTimeStorageType (bool isNullableInDatabase, Type dotNetType, IPropertyInformation? propertyInformation)
+    {
+      if (propertyInformation != null)
+      {
+        var storageTypeFromPropertyInfo = GetDateTimeStorageTypeFromPropertyInformation(propertyInformation, isNullableInDatabase, dotNetType);
+
+        if (storageTypeFromPropertyInfo != null)
+          return storageTypeFromPropertyInfo;
+      }
+
+      return new StorageTypeInformation(typeof(DateTime), "datetime2", DbType.DateTime2, isNullableInDatabase, null, dotNetType, new DefaultConverter(dotNetType));
```

Comment:
I would recommend flipping this. we get a positive condition and an exit condition all in one.
```suggestion
      if (propertyInformation == null)
      {
        return new StorageTypeInformation(typeof(DateTime), "datetime2", DbType.DateTime2, isNullableInDatabase, null, dotNetType, new DefaultConverter(dotNetType));
      }

      var storageTypeFromPropertyInfo = GetDateTimeStorageTypeFromPropertyInformation(propertyInformation, isNullableInDatabase, dotNetType);

      if (storageTypeFromPropertyInfo != null)
        return storageTypeFromPropertyInfo;
```

Reply:
This is actually not equivalent to the previous implementation as we now need to duplicate the new StorageTypeInformation as we now  need to duplicate the creation of the default.

Reply:
ah, yes, I see now. coding in a diff-editor is *hard* :D

Let's check the refactoring opportunities in the next PR when we do the default service.